### PR TITLE
feat: Update redis reserved ip range and Add deletion protection to database

### PR DESCRIPTION
**Pull request by venky-wandb**
Situation:
Screen Shot 2023-12-11 at 3 46 05 PM
When trying to install REDIS, it tries to establish a private services access connection with a shared VPC network. When creating this for Tower, somehow the reserved IP we have set for Redis appears to be in use. 

Solution:
Screen Shot 2023-12-11 at 3 44 56 PM
In order to get around that, I had to create the REDIS on a different reserved IP range than the hard-coded one we use. So this PR makes it a configurable parameter. [Test to show this works](https://app.terraform.io/app/weights-and-biases/workspaces/customer-tower/runs/run-wdS4eYAro94VUjGE)

Situation 2:
None of our GCP databases have deletion protection enabled because there are two different variables to control the deletion protection. More on the [thread here](https://weightsandbiases.slack.com/archives/C045T21276U/p1702325189940739?thread_ts=1702061366.060249&cid=C045T21276U)

Solution:
Screen Shot 2023-12-11 at 3 58 12 PM
Enable deletion protection at all levels (GCP, gcloud, CLI etc.) and not just at terraform deletion level. [Test to show this works](https://app.terraform.io/app/weights-and-biases/workspaces/customer-tower/runs/run-CsLLFXjLbV9ek8RS)

### CODE REVIEW

File: modules/database/main.tf [hunk 1]
```diff
@@ -34,6 +34,7 @@ resource "google_sql_database_instance" "default" {
     tier              = var.tier
     availability_type = var.availability_type
     user_labels       = var.labels
+    deletion_protection_enabled = true
```
Comment (nfoucha) [inline modules/database/main.tf:37]: terraform fmt is going to be grumpy about this one

**Review by nfoucha — CHANGES_REQUESTED** [review]

File: modules/database/main.tf (hunk 1 above)
Comment (venky-wandb) [reply modules/database/main.tf:37]: Yikes, I did a `terraform fmt` and confidently pushed. Let me retry that

Comment (venky-wandb) [reply modules/database/main.tf:37]: Ok, this should look better now.

File: modules/database/main.tf [hunk 2]
```diff
@@ -31,9 +31,10 @@ resource "google_sql_database_instance" "default" {
   deletion_protection = var.deletion_protection
 
   settings {
-    tier              = var.tier
-    availability_type = var.availability_type
-    user_labels       = var.labels
+    tier                        = var.tier
+    availability_type           = var.availability_type
+    user_labels                 = var.labels
+    deletion_protection_enabled = true
```
Comment (nfoucha) [inline modules/database/main.tf:37]: One other thing, we should probably have the value here use `var.deletion_protection`. I don't see a scenario where we'd want one enabled without the other so using the same variable should be ok.

Comment (venky-wandb) [reply modules/database/main.tf:37]: fair enough, I'll update that.

**Review by jsbroks — APPROVED** [review]

**Review by yogeshg — APPROVED** [review]
Tx for the detailed notes on testing and context.

**Review by nfoucha — APPROVED** [review]

**Comment by jsbroks** [comment]
This PR is included in version 1.22.0 :tada:

**Pull request merged**